packages/tinybased/src/lib/search/searched.ts

```diff
@@ -0,0 +1,101 @@
+/* eslint-disable @typescript-eslint/ban-types */
+import {
+  create,
+  getByID,
+  insert,
+  insertBatch,
+  Lyra,
+  remove,
+  search,
+  SearchParams,
+  SearchResult,
+} from '@lyrasearch/lyra';
+
+import { ObjectToCellStringType, TableBuilder } from '../TableBuilder';
+import { OnlyStringKeys, TinyBaseSchema } from '../types';
+
+type LyraInstances<TSchema extends TinyBaseSchema> = {
+  [K in OnlyStringKeys<TSchema>]: Lyra<ObjectToCellStringType<TSchema[K]>>;
+};
+export class Searched<TSchema extends TinyBaseSchema = {}> {
+  public static ERRORS = {
+    NO_TABLES_DEFINED: 'No tables defined for search',
+    TABLE_ALREADY_DEFINED: (table: string) => `Table ${table} already defined.`,
+    INDEX_NOT_CONFIGURED: (index: string) =>
+      `Index ${index} not configured for search`,
+    TABLE_NOT_IN_SCHEMA_BUILDER: (tableKeyToIndex: string) =>
+      `Table ${tableKeyToIndex} cannot be indexed. It is not defined for the SchemaBuilder.`,
+  };
+
+  public readonly tables: Map<OnlyStringKeys<TSchema>, TableBuilder<any, any>> =
+    new Map();
+
+  private indexes: LyraInstances<TSchema> = {} as LyraInstances<TSchema>;
+
+  async initialize() {
+    if (this.tables.size === 0) throw Error(Searched.ERRORS.NO_TABLES_DEFINED);
+    for (const [key, table] of this.tables) {
+      const instance = await create({
+        schema: table.schema,
+      });
+      this.indexes = { ...this.indexes, [key]: instance };
+    }
+    return this;
+  }
+
+  public addIndexedTable<
+    TName extends string,
+    TCells extends Record<string, unknown>
+  >(tableBuilder: TableBuilder<TName, TCells>) {
+    if (this.tables.has(tableBuilder.tableName as any)) {
+      throw Error(
+        Searched.ERRORS.TABLE_ALREADY_DEFINED(tableBuilder.tableName)
+      );
+    }
+    this.tables.set(tableBuilder.tableName as any, tableBuilder);
+
+    return this as unknown as Searched<TSchema & Record<TName, TCells>>;
+  }
+
+  public setTables<
+    TNewSchema extends TinyBaseSchema,
+    K extends OnlyStringKeys<TNewSchema>
+  >(tableBuilders: Map<K, TableBuilder<K, TNewSchema[K]>>) {
+    for (const [key, tableBuilder] of tableBuilders) {
+      this.tables.set(key as any, tableBuilder);
+    }
+    return this as unknown as Searched<TNewSchema>;
+  }
+
+  public insert<K extends OnlyStringKeys<TSchema>>(index: K, val: TSchema[K]) {
+    if (!this.indexes[index as any]) {
```

Comment:
Is this runtime check necessary? I would have thought that it would be impossible to call insert with a named index that isn't already tracked by the class